galleries/examples/text_labels_and_annotations/demo_annotation_box.py

```diff
@@ -1,12 +1,12 @@
 """
-===================
-AnnotationBbox demo
-===================
-
-`.AnnotationBbox` creates an annotation using an `.OffsetBox`, and
-provides more fine-grained control than `.Axes.annotate`.  This example
-demonstrates the use of AnnotationBbox together with three different
-OffsetBoxes: `.TextArea`, `.DrawingArea`, and `.OffsetImage`.
+======================
+Artists as annotations
+======================
+
+`.AnnotationBbox` creates an annotation using an `.OffsetBox` object, which is a class
+of container artists for positioning an artist relative to a parent artist. This allows
+for annotations that are texts, images, and arbitrary artists.  `.AnnotationBbox` also
+provides more fine-grained control than `.Axes.annotate`.
```

Comment:
Not sure whether my description is exactly correct, but I suggest to first state what you can achieve and only in the second step explain the technical background. I'd also push the discussion of `.Axes.annotate` down into the "Text" section.

```suggestion
`.AnnotationBbox` allows to use arbitrary Artists as annotations, including geometric shapes or images.

Technically, it accepts any `.OffsetBox` subclass, which is a container that supports positioning an artist relative to a parent artist.
```